stdlib/public/runtime/MetadataLookup.cpp

```diff
@@ -1893,6 +1893,267 @@ swift_stdlib_getTypeByMangledNameUntrusted(const char *typeNameStart,
                                     {}, {}).getType().getMetadata();
 }
 
+// ==== Function metadata functions ----------------------------------------------
+
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_SPI
+unsigned
+swift_func_getParameterCount(const char *typeNameStart, size_t typeNameLength) {
+  llvm::StringRef typeName(typeNameStart, typeNameLength);
+  for (char c : typeName) {
+    if (c >= '\x01' && c <= '\x1F')
+      return -1;
+  }
+
+  StackAllocatedDemangler<1024> demangler;
+  auto node = demangler.demangleSymbol(typeName);
+
+  if (!node) {
+    return -1;
+  }
+
+  switch (node->getKind()) {
+  case Node::Kind::Global:
+    if (node->getNumChildren() <= 0)
+      return -1;
+    node = node->getFirstChild();
+    break;
+  default:
+    return -2;
+  }
+
+  if (!node)
+    return -3;
+
+  switch (node->getKind()) {
```

Comment:
There is a lot of duplicate code here across _get* methods - could you please extract it into a `static` function?

Reply:
Yeah I'm going to clean these up a lot, just wanted to get it in your hands in "workable" condition so we can debug the _execute.

Thanks!

Reply:
I cleaned this up _a lot_ they all look quite nice now: https://github.com/apple/swift/pull/40605/commits/afe5a1d4d8f89967e433499d7632a5aa812c8cde

Could probably improve a bit more, but this is good enough.